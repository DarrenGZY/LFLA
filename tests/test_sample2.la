function othonomalising (vector[] bases, inspace ips, var n)
{
    vector vec;
    var i;
    for i = 1:n
    {
        vec = bases[i];
        var j;
        for j=1:i
        {
            vec = vec - ips<<bases[i], bases[j]>>*.bases[j];
        }
        bases[i]=vec/.(sqrt(ips<<vec, vec>>));
    }
}

function main()
{
    var n = 3;
    vector v1 = [1,2,3];
    vector v2 = [2.3, 2, 3];
    vector v3 = [2.3, 2, 3];

    vector bases[n] = {v1,v2, v3};
    
    vector result[n] = othonomalising(bases, n);
}
