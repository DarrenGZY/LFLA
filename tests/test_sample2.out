[array([ 0.26726124,  0.53452248,  0.80178373]), array([ 0.84847632, -0.52495657,  0.06714561]), array([ 0.45679247,  0.66234909, -0.59383022])]
